src/library/scala/util/Try.scala

```diff
@@ -174,6 +174,29 @@ sealed abstract class Try[+T] {
    *  `s` if this is a `Success`.
    */
   def transform[U](s: T => Try[U], f: Throwable => Try[U]): Try[U]
+
+  /**
+   * Returns `Left` with `Throwable` if this is a `Failure`, otherwise returns `Right` with `Success` value.
+   */
+  def toEither: Either[Throwable, T]
+
+  /**
+   * Applies `fa` if this is a `Failure` or `fb` if this is a `Success`.
+   *
+   * @example {{{
+   * val result: Try[Throwable, Int] = Try { string.toInt }
+   * log(result.fold(
+   *   ex => "Operation failed with " + ex,
+   *   v => "Operation produced value: " + v
+   * ))
+   * }}}
+   *
+   * @param fa the function to apply if this is a `Failure`
+   * @param fb the function to apply if this is a `Success`
+   * @return the results of applying the function
+   */
+  final def fold[U](fa: Throwable => U, fb: T => U): U = toEither.fold(fa, fb)
```

Comment:
This fails what I call the Bulletproof Principle: if you try to fold and the operation `fb` fails, you will get an exception instead of deferring to `fa`.  At the very least this would need to be carefully documented.  Also, implementing it this way forces one to go through an extra layer of wrappers in order to avoid having to write a simple match statement.  Let's just write the match statement!  (And, preferably, make it as safe as it reasonably could be.  Obviously if `fa` throws an exception, we wouldn't want to re-try it infinitely on that exception.)